$NetBSD: patch-aq,v 1.1 2001/03/23 15:57:46 agc Exp $

Patch from Wofgang Solfrank:

The changes to unproven-pthreads fix a problem with fstat not using
the pthreads filedescriptors, but the native ones (the code for this
was already there (slightly buggy), but it wasn't used for some
unknown reason).  In addition, the handling of select is improved, as
the original code may wake up not all threads waiting for some event.

--- pthreads/stat.c.orig	Sat Nov  6 23:30:50 1999
+++ pthreads/stat.c	Fri Mar 23 15:43:59 2001
@@ -54,7 +54,7 @@
 	int ret;
 
 	if ((ret = fd_lock(fd, FD_READ, NULL)) == OK) {
-		if ((ret = machdep_sys_fstat(fd_table[fd]->fd.i, buf)) < OK) {
+		if ((ret = machdep_sys_fstat(fd_table[fd]->fd, buf)) < OK) {
 			SET_ERRNO(-ret);
 			ret = NOTOK;
 		}
@@ -63,6 +63,7 @@
 	return(ret);
 }
 
+#ifdef	__notdef__		/* Not neccessary (for now) */
 /* ==========================================================================
  * stat()
  */
@@ -92,4 +93,4 @@
 	return(ret);
 
 }
-
+#endif
